### DISCUSSION TITLE
What about composable static website

### DISCUSSION THREAD

**Discussion by lahssenk** [nuejs/nue#247] [Q&A]
If I understand the docs correctly, there are currently a few advertised usecases for nue:
- content driven website, using markdown
- single page app with html, components and router

If I want to build a website for my company, have many static pages but reuse the same layout and write html not markdown, do I have an option? The argument would be to enjoy a composable structure and reuse the layout, but also using html instead of markdown and special syntax in order to stay closer to standard website building.

Am I off here ?

I know this does not align as much with "separation of concerns" between content, design, and application logic (JS), but it is still interesting for one that does not want to go with react and jsx and just simplify the codebase and DRY.

**Comment by tipiirai** [comment]
You pretty much nailed the idea of Nue. 

There is currently no such support for HTML layout reuse / override, but this is not a very hard thing to implement to the current codebase. Curious: why do you prefer HTML over Markdown for bulk content? 

Thanks!

**Comment by lahssenk** [comment]
Hi Tero!

What I like with markdown/nuemark:
- it is pretty lean
- it is very readable

What I don't like:
- if you come from another framework you have to unlearn a lot of things
- markdown is very strict (I created an app from template, run as is and it didn't build because of some spacing/indentation issue which is not easy to debug)
- as long as you want to add styling, components, images etc you transform yaml into a slim html (tags, classes, ids etc) which you won't use anywhere else than nue

This makes it a bit more of a hard sell for a company already using a framework (react, vue or else)

### First step

I build my first static page, it is almost as easy as pure html + css:

- index.html
- base.css
- site.yaml

### Second step

I want to add a second page, and reuse things:

- components.nue
- layout.html
- index.html
- site.yaml
- contact/
- contact/index.html
- (contact/layout.yaml)
- contact/app.yaml


### Why not:
- components.nue (reusable parts, some not in the layout)
- layout.html (avoid writing header, footer everytime)
- site.yaml (data and config)
- index.html (a page)
- contact.html (another page)
- about.html (another one)

**Comment by tipiirai** [comment]
I see your point and I have already planned to implement something like the above.

I'd say that Markdown is more of a standard thing in static site generators and for content-heavy websites. Injecting content inside the HTML file makes the site harder to maintain, and makes it easier to break. It's also harder to centralize design into a external design system that all the "standardized" content-files apply to. The HTML-based system has likely more to learn when coming from another framework.
